pandas/core/indexes/base.py

```diff
@@ -6037,7 +6038,7 @@ def _get_indexer_strict(self, key, axis_name: str_t) -> tuple[Index, np.ndarray]
         if isinstance(key, Index):
             # GH 42790 - Preserve name from an Index
             keyarr.name = key.name
-        if keyarr.dtype.kind in "mM":
+        if keyarr.dtype.kind in "mM" and not isinstance(keyarr.dtype, ArrowDtype):
```

Comment:
i think use `lib.is_np_dtype(keyarr.dtype, "mM")`?

Reply:
I would have to include `DatetimeTZDtype` here then too based on failing tests so is `lib.is_np_dtype(keyarr.dtype, "mM") or isinstance(keyarr.dtype, DatetimeTZDtype)` better?

Reply:
good point.  yes, i think that would be more explicit about the cases that are supposed to go through here